### ISSUE TITLE
No browser found in android when trying to login

### ISSUE THREAD

**Issue by daald** [zulip/zulip-mobile#5466]
On my Samsung Galaxy S10 I get a "no browser found" error when trying to login (server is chat.sac-uto.ch).

Phone is up-to-date:
Android 12
patchlevel May 1, 2022

Packages installed:
from the releases page: v27.189 https://github.com/zulip/zulip-mobile/releases/download/v27.189/app-arm64-v8a-release.apk

**Comment by gnprice** [comment]
Thanks @daald for the report! Glad it's now working for you.

Can you say a bit more about how you disabled Chrome? We've heard about an issue like this once before, and would love to better understand what's going on -- both to help the next person who sees it, and ideally so we can fix the problem entirely.

**Comment by daald** [comment]
Sure.
1. Long-press to Chrome in the launcher
2. press the (i)
3. on the bottom, there is a "disable" button, near "force stop" and "open"

This is to disable apps which are included in the firmware image, like youtube or spotify. if an app was installed through apk or a store, there is an "uninstall" button instead.

**Comment by zer0def** [comment]
To expand on observed behavior:
- [ReactApplicationContext is unable to find the web browser explicitly chosen in Android as the default](https://github.com/zulip/zulip-mobile/blob/e433197cff964416c5462075cad44fd451751ada/android/app/src/main/java/com/zulipmobile/CustomTabsAndroid.java#L46)
- as soon as [Google Chrome (`com.android.chrome`) is present in the system](https://github.com/zulip/zulip-mobile/blob/e433197cff964416c5462075cad44fd451751ada/android/app/src/main/java/com/zulipmobile/CustomTabsHelper.java#L16-L23), the [login portal's URL is opened properly](https://github.com/zulip/zulip-mobile/blob/e433197cff964416c5462075cad44fd451751ada/android/app/src/main/java/com/zulipmobile/CustomTabsAndroid.java#L40) **within the system-wide default browser** (which, in my particular case, is **not** Chrome)

**Comment by gnprice** [comment]
Thanks @daald for those details, and @zer0def for the debugging!

Looking at the code of ours that's at those links, the logic definitely seems dubious — particularly in how it checks for `com.android.chrome` by name before launching a CustomTabsIntent, which isn't specific to Chrome. I'm not seeing how that bug would lead to a situation where *dis*abling Chrome works around the problem, but I'm glad it did for @daald.

Because this app is now in maintenance mode as we're focused on building [the new Zulip mobile app in Flutter](https://chat.zulip.org/#narrow/stream/2-general/topic/Flutter/near/1708728), this is an issue I don't expect us to go back and fix for the legacy app. In the new app, fortunately, this functionality is provided by a package from Flutter upstream, and I think its logic is more solid than this. So I'd expect this sort of issue shouldn't arise in the new app — and if it does, I'll definitely be interested in getting to the bottom of it there.

**Comment by daald** [comment]
No, it was the other way round. I had chrome disabled, that's why Zulip failed. After re-enabline, it worked. But there were 2 other browsers installed, so there was no reason to fail, just because Chrome was not there.
The above instructions where for reproducing, not for solving the issue...

**Comment by gnprice** [comment]
Ah excellent, thanks for clarifying. All makes sense then. (Still a bug, but it now makes sense how the bug caused the pattern of behavior you saw.)

The reason I'd thought that disabling Chrome was the workaround was this part:
> solved. I've disabled chrome browser

**Comment by gnprice** [comment]
I've edited the title a bit to reflect the debugging above, and also the alternate symptom reported in #5795 that I believe is caused by this same issue.
